assets/src/components/stepperBar.tsx

```diff
@@ -0,0 +1,187 @@
+import useComponentSize from "@rehooks/component-size"
+import React, { useRef } from "react"
+import { SvgProps } from "../helpers/bsIcons"
+
+export const StepperBar = ({
+  totalSteps,
+  currentStep,
+}: {
+  totalSteps: number
+  currentStep: number
+}) => {
+  const stepperRef = useRef(null)
+  const { width } = useComponentSize(stepperRef)
+
+  const stepWidth = width / totalSteps
+
+  let stepper
+  if (currentStep == totalSteps) {
+    stepper = <FilledStepper stepWidth={stepWidth} stepCount={currentStep} />
+  } else {
+    stepper = (
+      <>
+        <FilledStepper
+          stepWidth={stepWidth}
+          stepCount={currentStep}
+          isPartial={true}
+        />
+        <UnfilledStepper
+          stepWidth={stepWidth}
+          stepCount={totalSteps - currentStep}
+        />
+      </>
+    )
+  }
+
+  return (
+    <div className="c-stepper-bar d-grid" ref={stepperRef}>
+      <StepperBusIcon style={{ marginLeft: stepWidth * currentStep - 66 }} />
+      <div className="d-flex">{stepper}</div>
+    </div>
+  )
+}
+
+const barHeight = 8
+const grayFill = "#D4D7DB"
+const purpleFill = "#6D39AC"
+
+const FilledStepper = ({
+  stepWidth: stepWidth,
+  stepCount: stepCount,
+  isPartial: isPartial,
+}: {
+  stepWidth: number
+  stepCount: number
+  isPartial?: boolean
+}) => (
+  <svg width={stepWidth * stepCount} height={barHeight}>
+    {isPartial && (
+      <rect
+        width={stepWidth * stepCount}
+        height={barHeight}
+        x={barHeight}
+        fill={grayFill}
+      />
+    )}
+    <rect
+      width={stepWidth * stepCount}
+      height={barHeight}
+      rx={barHeight / 2}
+      ry={barHeight / 2}
+      fill={purpleFill}
+    />
+    <StopDots stepWidth={stepWidth} numDots={stepCount} />
+  </svg>
+)
+
+const UnfilledStepper = ({
+  stepWidth: stepWidth,
+  stepCount: stepCount,
+}: {
+  stepWidth: number
+  stepCount: number
+}) => (
+  <svg width={stepWidth * stepCount} height={barHeight}>
+    <rect width={barHeight} height={barHeight} fill={grayFill} />
+    <rect
+      width={stepWidth * stepCount}
+      height={barHeight}
+      rx={barHeight / 2}
+      ry={barHeight / 2}
+      fill={grayFill}
+    />
+    <StopDots stepWidth={stepWidth} numDots={stepCount} />
+  </svg>
+)
+
+const StopDots = ({
+  stepWidth: stepWidth,
+  numDots: numDots,
+}: {
+  stepWidth: number
+  numDots: number
+}) => {
+  let dots = []
```

Comment:
The build is being ornery about this because `dots` can be declared as a `const` here (technically I guess you can add things to a `const` array 🤯).

Reply:
Ahh yep! I do not make any changes to the variable itself

```suggestion
  const dots = []
```